src/main/java/journeymap/client/data/WorldData.java

```diff
@@ -285,10 +287,25 @@ public static String getSafeDimensionName(WorldProvider worldProvider)
 
         try
         {
-            return worldProvider.getDimensionName();
+            if (dimNames == null) {
+                dimNames = new TIntObjectHashMap<String>();
+            }
+            if (dimNames.containsKey(worldProvider.dimensionId)) {
+                return dimNames.get(worldProvider.dimensionId);
+            }
+
+            String langKey = String.format("jm.dimension.%1$d.name", worldProvider.dimensionId);
+            String dimName = Constants.getString(langKey);
+            if (dimName.equals(langKey))
+            {
+                dimName = worldProvider.getDimensionName();
+            }
+
+            return dimNames.put(worldProvider.dimensionId, dimName);
         }
         catch (Exception e)
         {
+            JMLogger.logOnce(String.format("Failed to retrieve dimension %d name: error: %s", worldProvider.dimensionId, e), e);
```

Comment:
this can be
`JMLogger.logOnce(String.format("Failed to retrieve dimension %d name: error: ", worldProvider.dimensionId), e);`